### PR TITLE
Feature/contact

### PR DESCRIPTION
**Pull request by msomierick**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    name='wagtail-contact',
+    version='0.1',
+    packages=find_packages(),
+    include_package_data=True,
+    license='BSD License',  # example license
+    description='A simple Wagtail contact app.',
+    long_description=README,
+    url='https://github.com/regulusweb/wagtail-contact',
+    author='Regulus web',
+    author_email='erick@regulusweb.com',
```
Comment (craigloftus) [inline setup.py]: @solarissmoke  Should use `info@regulusweb.com` here?

File: wagtailcontact/admin.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
```
Comment (craigloftus) [inline wagtailcontact/admin.py]: Can drop this file as well.

File: wagtailcontact/forms.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            reply_to=reply_to
+        )
+        msg.send()
+
+    def get_to(self, page):
+        try:
+            return [page.enquiry_email]
+        except AttributeError:
+             return [e for _, e in settings.MANAGERS]
+
```
Comment (craigloftus) [inline wagtailcontact/forms.py]: Few too many new lines in here.

File: wagtailcontact/forms.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            message,
+            from_email,
+            to,
+            reply_to=reply_to
+        )
+        msg.send()
+
+    def get_to(self, page):
+        try:
+            return [page.enquiry_email]
```
Comment (craigloftus) [inline wagtailcontact/forms.py]: This mixin needs to provide an `enquiry_email` field.

File: setup.py (hunk 1 above)
Comment (solarissmoke) [reply setup.py]: Yes, please use info.

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    install_requires=[
+        'django-honeypot==0.6.0'
+    ],
+    classifiers=[
+        'Environment :: Web Environment',
+        'Framework :: Django',
+        'Framework :: Django :: 1.11',  # replace "X.Y" as appropriate
```
Comment (solarissmoke) [inline setup.py]: The comments in this list should be removed.

File: wagtail_contact.egg-info/dependency_links.txt [hunk 6]
```diff
@@ -0,0 +1 @@
+
```
Comment (solarissmoke) [inline wagtail_contact.egg-info/dependency_links.txt]: Why is this file needed?

Comment (solarissmoke) [reply wagtail_contact.egg-info/dependency_links.txt]: Never mind - as Craig noted this whole directory shouldn't be committed.

File: wagtailcontact/forms.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+
+class ContactForm(forms.Form):
+
+    subject_template = "wagtailcontact/email/subject.txt"
+    txt_template = "wagtailcontact/email/message.txt"
+    success_url = None
+
+    name = forms.CharField(max_length=100, required=True)
+    email = forms.EmailField(max_length=200, required=True)
```
Comment (solarissmoke) [inline wagtailcontact/forms.py]: `EmailField` comes with a default `max_length`, don't think you need to specify your own.

File: wagtailcontact/mixins.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    def get_context(self, request):
+        ctx = super().get_context(request)
+        ctx['form'] = self.form
+        return ctx
+
+    def get_success_url(self):
+        return self.success_url or self.url
+
+    def get_success_message(self):
+        msg = 'Thank you! We will get back to you as soon as possible.'
```
Comment (solarissmoke) [inline wagtailcontact/mixins.py]: This is redundant - just set this as the default in `self.success_message`.

File: setup.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+os.chdir(os.path.normpath(os.path.join(os.path.abspath(__file__), os.pardir)))
+
+setup(
+    name='wagtail-contact',
+    version='0.1',
+    packages=find_packages(),
+    include_package_data=True,
+    license='BSD License',  # example license
+    description='A simple Wagtail contact app.',
+    long_description=README,
```
Comment (solarissmoke) [inline setup.py]: This irks me a bit - where did you get this pattern from?

File: setup.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    url='https://github.com/regulusweb/wagtail-contact',
+    author='Regulus web',
```
Comment (solarissmoke) [inline setup.py]: `author='Regulus Ltd'`

File: setup.py (hunk 9 above)
Comment (msomierick) [reply setup.py]: Sorry, the whole setup.py or how I read the README.md file?
I just made a few changes to example given here: https://docs.djangoproject.com/en/1.11/intro/reusable-apps/

Comment (solarissmoke) [reply setup.py]: How you read the README file - but apparently it seems to be quite common. Leave it as is for now.

File: wagtailcontact/forms.py (hunk 4 above)
Comment (msomierick) [reply wagtailcontact/forms.py]: This seems interesting, adding the field requires us to add it to Page.content_panels(). Doing this without inheriting `Page` fails.
@solarissmoke will it be ok to have a `ContactPage` which inherits from this mixin and `Page`, which can then be used by subclasses?

Comment (solarissmoke) [reply wagtailcontact/forms.py]: The problem with subclassing a `Page` model is that the model is not abstract, and you will end up with multiple table joins just to fetch a single page object.

You should not need to have to subclass page, however. Just set `content_panels` = `Page.content_panels + [...]`. Without subclassing.

It will mean you have to add Wagtail to your dependencies (maybe don't pin a version though).

**Comment by solarissmoke** [comment]
This looks good to me - just need some tests now.

File: tests/test_forms.py [hunk 11]
```diff
@@ -0,0 +1,34 @@
+from django.core import mail
+from django.test import TestCase
+
+from wagtailcontact.forms import ContactForm
+
+
+class ContactFormTestCase(TestCase):
```
Comment (solarissmoke) [inline tests/test_forms.py]: Have you tested the `save()` method to ensure it does what it is supposed to? Same for the `get_to()` method.
